### ISSUE TITLE
LSM2 forward_features_full crashes when inherited_mask uses documented default

### ISSUE THREAD

**Issue by NarayanSchuetz** [AshleyLab/OpenMHC#12]
## Problem

`LSM2ViT1D.forward_features_full` declares `inherited_mask: torch.Tensor | None = None`, but immediately passes it into `forward_encoder(x, inherited_mask)`. `forward_encoder` expects a tensor and does not handle `None`, so calling `forward_features_full(x)` with the documented default crashes.

This path appears to be dead code for the current imputation benchmark, but it is a broken public/internal API for dense feature extraction.

## Implications

Current benchmark results should not be affected if this method is unused. Future downstream feature extraction, segmentation, or probing code that calls `forward_features_full(x)` without an explicit mask will fail unexpectedly.

## Proposed Fix

Make `forward_features_full` honor its default argument by creating an all-zero inherited mask when `inherited_mask is None`:

```python
if inherited_mask is None:
    B = x.shape[0]
    inherited_mask = torch.zeros((B, self.num_patches), device=x.device)
```

Add a regression test that calls `forward_features_full(x)` on a tiny model and asserts finite dense features with shape `(B, num_patches, decoder_embed_dim)`.
